src/python/pants/option/options_bootstrapper.py

```diff
@@ -93,11 +93,16 @@ def register_global(*args, **kwargs):
         return bootstrap_options
 
     @classmethod
-    def create(cls, env: Mapping[str, str], args: Sequence[str]) -> "OptionsBootstrapper":
+    def create(
+        cls, env: Mapping[str, str], args: Sequence[str], *, allow_pantsrc: bool = False
```

Comment:
It's possible that this param should not have a default, to make it harder to accidentally leave off?

Reply:
Yes, I think so. I could imagine subtle bugs where, in production, we accidentally don't override this value.

Reply:
Also most tests seem to use the testutil `create_options_bootstrapper()`, which can continue to always set `allow_pantsrc=False`.